src/components/welcome/MyAccounts/PaginatedSafeList.tsx

```diff
@@ -46,8 +50,15 @@ const PaginatedSafeList = ({ safes, title, action, onLinkClick }: PaginatedSafeL
       </div>
       {shownSafes.length ? (
         shownSafes.map((item) => <AccountItem onLinkClick={onLinkClick} {...item} key={item.chainId + item.address} />)
+      ) : !wallet ? (
+        <Typography variant="body2" color="text.secondary" textAlign="center" my={3} mx="auto" width={250}>
+          Connect a wallet to view your Safe Accounts or to create a new one
+          <Button onClick={handleConnect} disableElevation size="small" sx={{ mt: 2 }}>
+            Connect a wallet
+          </Button>
+        </Typography>
```

Comment:
Should it be here or perhaps in the parent component?

Reply:
you mean replace the 'create account' button with 'connect wallet' button when the wallet is not connected?
True that it doesn't make sense to have a create account option if you're not connected. All it does is redirect to the welcome page

Reply:
I meant just architecturally that it would be better to keep only the list iself in the PaginatedSafeList. But your comment makes even more sense.

Reply:
ok I changed it to pass the empty message as a prop from the parent